doc/api/child_process.markdown

```diff
@@ -1,359 +1,107 @@
-signal to a child process that has already exited is not an error but may
-have unforeseen consequences. Specifically, if the process identifier (PID) has
-been reassigned to another process, the signal will be delivered to that
-process instead. What happens next is anyone's guess.
-
-Note that while the function is called `kill`, the signal delivered to the
-child process may not actually kill it.  `kill` really just sends a signal
-to a process.
-
-See `kill(2)`
-
-### child.pid
-
-* {Integer}
-
-The process identifier (PID) of the child process.
-
-Example:
-
-    const spawn = require('child_process').spawn;
-    const grep = spawn('grep', ['ssh']);
-
-    console.log(`Spawned child pid: ${grep.pid}`);
-    grep.stdin.end();
-
-### child.send(message[, sendHandle][, callback])
-
-* `message` {Object}
-* `sendHandle` {Handle object}
-* `callback` {Function}
-* Return: Boolean
-
-When using [`child_process.fork()`][] you can write to the child using
-`child.send(message[, sendHandle][, callback])` and messages are received by
-a `'message'` event on the child.
-
-For example:
-
-    const cp = require('child_process');
-    const n = cp.fork(`${__dirname}/sub.js`);
-
-    n.on('message', (m) => {
-      console.log('PARENT got message:', m);
+    ls.stderr.on('data', (data) => {
+      console.log(`stderr: ${data}`);
     });
 
-    n.send({ hello: 'world' });
-
-And then the child script, `'sub.js'` might look like this:
-
-    process.on('message', (m) => {
-      console.log('CHILD got message:', m);
+    ls.on('close', (code) => {
+      console.log(`child process exited with code ${code}`);
     });
 
-    process.send({ foo: 'bar' });
-
-In the child, the `process` object will have a `send()` method, and `process`
-will emit objects each time it receives a message on its channel.
-
-There is a special case when sending a `{cmd: 'NODE_foo'}` message. All messages
-containing a `NODE_` prefix in its `cmd` property will not be emitted in
-the `'message'` event, since they are internal messages used by Node.js core.
-Messages containing the prefix are emitted in the `'internalMessage'` event.
-Avoid using this feature; it is subject to change without notice.
-
-The `sendHandle` option to `child.send()` is for sending a TCP server or
-socket object to another process. The child will receive the object as its
-second argument to the `'message'` event.
+By default, pipes for `stdin`, `stdout` and `stderr` are established between
+the parent Node.js process and the spawned child. It is possible to stream data
+through these pipes in a non-blocking way. *Note, however, that some programs
+use line-buffered I/O internally. While that does not affect Node.js, it can
+mean that data sent to the child process may not be immediately consumed.*
+
+For convenience, the `child_process` module provides a handful of synchronous
+and asynchronous alternatives to `child_process.spawn()`, each of which are
+documented fully [below][]:
```

Comment:
I think its worth pointing out that all the higher-level methods are themselves implemented in terms of spawn or spawnSync. I think you are kindof implying it, but not outright saying it. Its easier to understand fork/exec/execFile if you know that they are just wrappers around spawn(), and they cannot at the end of day do anything spawn can't.